app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/optionsStepForm/optionsBlock/optionsBlock.jsx

```diff
@@ -89,8 +92,43 @@ export const OptionsBlock = ({
         });
       options.push(...optionalOptions);
     }
+    historyItems.length > 0 &&
+      historyItems.some(
+        (item) =>
+          item.issue && TIDefectsGroup.find((type) => type.locator === item.issue.issueType),
+      ) &&
+      options.push({
+        ownValue: ALL_LOADED_TI_FROM_HISTORY_LINE,
+        label: {
+          id: ALL_LOADED_TI_FROM_HISTORY_LINE,
+          defaultMessage: formatMessage(messages.allLoadedTIFromHistoryLine),
+        },
+      });
     return options;
   };
+  const setItemsFromHistoryLine = (value) => {
+    const preparedHistoryItems = [
+      currentTestItem,
+      ...historyItems
+        .reduce((items, item) => {
+          const currentDefectType =
+            (item.issue && TIDefectsGroup.find((type) => type.locator === item.issue.issueType)) ||
+            {};
+          return currentDefectType.typeRef === TO_INVESTIGATE.toUpperCase() &&
+            item.issue &&
+            item.id !== currentTestItem.id
+            ? [...items, { ...item, itemId: item.id }]
+            : items;
```

Comment:
Since `item.issue` appears in two conditions, it should be taken out in a separate check, this will simplify the perception of the entire code.
You could also take out `item.id !== currentTestItem.id`.
Something like
`if (!item.issue && item.id !== currentTestItem.id) {`
           `return items;`
        `  }`
          `const currentDefectType = TIDefectsGroup.find((type) => type.locator === item.issue.issueType) || {};`
         ` return currentDefectType.typeRef === TO_INVESTIGATE.toUpperCase()`
           ` ? [...items, { ...item, itemId: item.id }]`
           ` : items;`